app/services/datastore/application_rehydration.rb

```diff
@@ -28,10 +28,10 @@ def call # rubocop:disable Metrics/MethodLength, Metrics/AbcSize, Metrics/Cyclom
         outgoings_payments: parent.outgoings&.outgoings_payments || [],
         documents: parent.documents,
         additional_information: parent.additional_information,
-        income_payments: parent.income&.income_payments,
-        income_benefits: parent.income&.income_benefits,
-        employments: parent.income&.employments,
-        businesses: parent.income&.businesses,
+        income_payments: parent.income&.income_payments || [],
+        income_benefits: parent.income&.income_benefits || [],
+        employments: parent.income&.employments || [],
+        businesses: parent.income&.businesses || [],
```

Comment:
I've added integration specs that use the submitted application json as the response from the datastore. Default values for these are added by the datastore and are missing here. I'd rather the datastore wasn't doing any of that so fixing how we handle it here.